src/components/Card/TodosCard/TodosCard.tsx

```diff
@@ -0,0 +1,272 @@
+      try {
+        await TodoService.update(todoToUpdate);
+
+        setTodos(oldArray =>
+          oldArray.map(todo => {
+            if (todo.id === todoToUpdate.id) {
+              if (todo.deadlineTime !== todoToUpdate.deadlineTime) {
+                remaindersWorker.postMessage({
+                  type: "update",
+                  data: todoToUpdate,
+                });
+              }
+              return todoToUpdate;
+            }
+            return todo;
+          })
+        );
+      } catch (error) {
+        fireSwalError("The todo couldn't be updated");
+      }
+    },
+    [remaindersWorker]
+  );
+
+  const handleFilterChange = useCallback(
+    (event: React.MouseEvent<HTMLElement>, newFilter: string) => {
+      setTodosFilter(newFilter);
+    },
+    []
+  );
+
+  // TODO: the name doesn't imply boolean result, but a filter action - rename
+  // ANSWER: I'm not sure which is better "toShowTodo" or "isTodoFiltered"
+  const toShowTodo = (todoToFilter: Todo): boolean => {
+    return todosFilter === Filter.ALL
+      ? true
+      : (todoToFilter.isCompleted ? Filter.COMPLETED : Filter.ACTIVE) ===
+          todosFilter;
+  };
+
+  const openTimeModal = useCallback((todo: Todo) => {
+    setTimeModalState(true);
+    setTodoToUpdate({ ...todo });
+  }, []);
+
+  return (
+    <React.Fragment>
+      <Box m="auto" width="50%">
+        <Card>
+          <TodoInput onSubmit={handleTodoSubmit} />
+          <List>
+            {S.pipe([
+              S.filter(toShowTodo),
+              S.map((todo: Todo) => (
+                // TODO: if you're not familiar with react reconcilation, then please read (https://reactjs.org/docs/reconciliation.html) + read about react diff alogirthm from the same source
+                // and explain why we need the `key` prop
+                // ANSWER: rendering a list isn't easy for react diff algorithm, if an element is added at the end of the list
+                // then it's easy, react diff algo will compare all elements and find out they are the same, but just one element was added to the end.
+                // When an element is added at the start of the list, react will have a hard time finding out that the rest of the elements are the same,
+                // and finding this out is pretty expensive. So a solution react came up with is binding a key to each element rendered from a list, that way
+                // react diff algo will be able to know if elements position just moved, or their state changed.
+                <TodoItem
+                  key={todo.id}
+                  todo={todo}
+                  onDelete={handleTodoDelete}
+                  onTodoUpdate={handleTodoUpdate}
+                  openTimeModal={openTimeModal}
+                />
+              )),
+            ])(todos)}
+          </List>
+          <CardFooter
+            todosFilter={todosFilter}
+            numberOfUncompletedTodos={useMemo(getNumberOfUncompletedTodos, [
```

Comment:
that 's not the way you should use `useMemo`, make it outside the return statement and assign it to a variable.
that way `useMemo` will be invoked every return and you achieve nothing by doing so